eeraster.py

```diff
@@ -130,6 +132,104 @@ def values_at_df(self, coordinates:pd.DataFrame, column_name: str = "value") ->
       all_dfs = _getPool().starmap(_query_mp, query_list)
       return pd.concat(all_dfs, ignore_index=True)
 
+def stamp_isoscape(ee_asset_path: str, properties: Dict[str, typing.Any]):
+  """
+  stamp_isoscape function
+  ------------------------
+  This function stamps an Earth Engine isoscape with properties.
+  This function can also be used to edit an existing stamp.
+  ------------------------
+  Parameters:
+  ee_asset_path : str
+      GeoTIFF filename in Earth Engine (with the full path) of the isoscape to be stamped
+      e.g: projects/river-sky-386919/assets/isoscapes/d18O_isoscape
+  properties: Dict[str, Any]
+      A dictionary of metadata to apply to the Earth Engine asset
+      e.g. {"p_value" : 0.95}
+  """
+  eeddf.initialize_ddf()
+  isoscape = ee.Image(f"{EE_ASSET_FOLDER}/{ee_asset_path}")
+  isoscape.set(var_args=properties)
+
+def show_stamps(ee_asset_path: str):
+  
+  """
+  show_stamps function
+  ------------------------
+  This function returns all the properties on an Earth Engine isoscape
+  ------------------------
+  Parameter:
+  ee_asset_path : str
+      GeoTIFF filename in Earth Engine (with the full path) of the isoscape to be stamped
+      e.g: projects/river-sky-386919/assets/isoscapes/d18O_isoscape
+  """
+  eeddf.initialize_ddf()
+  isoscape = ee.Image(ee_asset_path)
+  return isoscape.getInfo().properties()
+
+def del_stamp(filename: str, property_name: str):
+  """
+  del_stamp function
+  ------------------------
+  This function deletes a property from an Earth Engine isoscape
+  ------------------------
+  Parameters:
+  filename : str
+      GeoTIFF filename (with the full path) of the stamped isoscape
+      e.g: projects/river-sky-386919/assets/isoscapes/d18O_isoscape
+  property_name : str
+      Name of the metadata to be deleted
+      e.g: "p_value"
+  """  
+  eeddf.initialize_ddf()
+  isoscape = ee.Image(e_asset_path)
+  isoscape.set(property_key, None)
+
+def ingest_isoscape(
+    isoscape_path: str,
+    properties: Dict[str, typing.Any],
```

Comment:
you should keep greta's version and just read the metadata from the isoscape file instead of being passed the properties dict.
This ensures what nicholas suggested in the meeting (which is really a good idea) - that every file we run thru validation ends up having stamped metadata.
And it allows someone like Dr. Martinelli to run this script to upload a validated isoscape to earth engine without having to modify the file (or rerun the entire validation pipeline).